src/uPlot.js

```diff
@@ -1868,11 +1868,55 @@ export default function uPlot(opts, data, then) {
 	}
 
 	function mouseLeave(e, src, _x, _y, _w, _h, _i) {
-		if (!cursor.locked && !dragging) {
+		if (!cursor.locked) {
+			let _dragging = dragging;
+
+			if (dragging) {
+				// handle case when mousemove aren't fired all the way to edges by browser
+				let dLft = mouseLeft1;
+				let dRgt = plotWidCss - mouseLeft1;
+				let dTop = mouseTop1;
+				let dBtm = plotHgtCss - mouseTop1;
+
+				let xProx = true;
+				let yProx = true;
+				let proxThresh = 10;
+
+				if (dragX && dragY) {
+					// maybe omni corner snap
+					xProx = mouseLeft1 <= proxThresh || mouseLeft1 >= plotWidCss - proxThresh;
+					yProx = mouseTop1  <= proxThresh || mouseTop1  >= plotHgtCss - proxThresh;
+				}
+				
+				if (dragX && xProx) {	
+					let xMin = min(dLft, dRgt);
+					if (xMin == dLft)
```

Comment:
please make sure any leading `if ()` blocks have an empty line before them. more legible to me than having `let` immediately adjacent, especially with identical alignment.